go/ql/lib/semmle/go/security/FileSystemAccess.qll

```diff
@@ -1,211 +1,217 @@
 import go
 
 /**
- * Provide File system access sinks of [fasthttp](https://github.com/valyala/fasthttp) web framework
+ * The File system access sinks of [fasthttp](https://github.com/valyala/fasthttp) web framework
  */
 class FastHttpFileSystemAccess extends FileSystemAccess::Range, DataFlow::CallNode {
+  int pathArg;
+
   FastHttpFileSystemAccess() {
-    exists(DataFlow::Method mcn |
+    exists(Method m |
       (
-        mcn.hasQualifiedName("github.com/valyala/fasthttp.RequestCtx", ["SendFileBytes", "SendFile"]) or
-        mcn.hasQualifiedName("github.com/valyala/fasthttp.Response", ["SendFile"])
+        m.hasQualifiedName(package("github.com/valyala/fasthttp", ""), "RequestCtx",
+          ["SendFileBytes", "SendFile"])
+        or
+        m.hasQualifiedName(package("github.com/valyala/fasthttp", ""), "Response", "SendFile")
       ) and
-      this = mcn.getACall()
+      this = m.getACall() and
+      pathArg = 0
     )
     or
-    exists(DataFlow::Function f |
-      f.hasQualifiedName("github.com/valyala/fasthttp",
+    exists(Function f |
+      f.hasQualifiedName(package("github.com/valyala/fasthttp", ""),
         [
           "ServeFile", "ServeFileUncompressed", "ServeFileBytes", "ServeFileBytesUncompressed",
           "SaveMultipartFile"
         ]) and
-      this = f.getACall()
+      this = f.getACall() and
+      pathArg = 1
     )
   }
 
-  override DataFlow::Node getAPathArgument() {
-    this.getTarget().getName() =
-      [
-        "ServeFile", "ServeFileUncompressed", "ServeFileBytes", "ServeFileBytesUncompressed",
-        "SaveMultipartFile"
-      ] and
-    result = this.getArgument(1)
-    or
-    this.getTarget().getName() = ["SendFile", "SendFileBytes"] and
-    result = this.getArgument(0)
-  }
+  override DataFlow::Node getAPathArgument() { result = this.getArgument(pathArg) }
 }
 
 /**
- * Provide File system access sinks of `net/http` package
+ * The File system access sinks of `net/http` package
  */
 class HttpServeFile extends FileSystemAccess::Range, DataFlow::CallNode {
+  int pathArg;
+
   HttpServeFile() {
-    exists(DataFlow::Function mcn |
-      mcn.hasQualifiedName("net/http", "ServeFile") and
-      this = mcn.getACall()
+    exists(Function f |
+      f.hasQualifiedName("net/http", "ServeFile") and
+      this = f.getACall() and
+      pathArg = 2
     )
   }
 
-  override DataFlow::Node getAPathArgument() { result = this.getArgument(2) }
+  override DataFlow::Node getAPathArgument() { result = this.getArgument(pathArg) }
```

Comment:
No need to use a field to store the path arg in this class, since it is always the same. You can just hardcode it as 2.